JSDemos/Demos/TagBox/GroupedItems/description.md

```diff
@@ -1 +1,9 @@
-The TagBox component allows you to&nbsp;group items within the drop-down list. This demo illustrates how to&nbsp;enable grouping and how to&nbsp;define a&nbsp;custom template for groups.
\ No newline at end of file
+The TagBox component can display items as a two-level hierarchy: parent and child items. To group items, set the [grouped](/Documentation/ApiReference/UI_Components/dxTagBox/Configuration/#grouped) property to **true** and ensure the data source defines the hierarchy in one of the following ways:
+
+- **Grouping Field**    
+The data source can contain a plain array of objects with multiple fields. Assign a field to the DataSource's [group](/Documentation/ApiReference/Data_Layer/DataSource/Configuration/#group) property to create a hierarchical display. Parent items display unique values from the specified field, and Child items display grouped values. This demo shows how to use this type of data source.
+
+- **Hierarchical Data Source Structure**    
+You can use a [dataSource](/Documentation/ApiReference/UI_Components/dxTagBox/Configuration/#dataSource) where each entry is an object that contains the **key** and **items** fields. The **key** field specifies the group header (parent), and the **items** field holds an array of child items. A data source can specify different field names if the data is structured in the same way. In this case, implement the [DataSource](/Documentation/ApiReference/Data_Layer/DataSource/)'s [map](/Documentation/ApiReference/Data_Layer/DataSource/Configuration/#map) function to create **key** and **items** field mappings. Refer to the following help topic for more information: [Item Mapping](/Documentation/Guide/Data_Binding/Data_Layer/#Reading_Data/Data_Transformation/Item_Mapping).
```

Comment:
It's better to address the data source as a term rather than an API reference item. This also allows us to preserve the indefinite article.

```suggestion
You can use a [data source](/Documentation/ApiReference/UI_Components/dxTagBox/Configuration/#dataSource) where each entry is an object that contains the **key** and **items** fields. The **key** field specifies the group header (parent), and the **items** field holds an array of child items. A data source can specify different field names if the data is structured in the same way. In this case, implement the [DataSource](/Documentation/ApiReference/Data_Layer/DataSource/)'s [map](/Documentation/ApiReference/Data_Layer/DataSource/Configuration/#map) function to create **key** and **items** field mappings. Refer to the following help topic for more information: [Item Mapping](/Documentation/Guide/Data_Binding/Data_Layer/#Reading_Data/Data_Transformation/Item_Mapping).
```

Reply:
This text was borrowed from a [demo](https://js.devexpress.com/Demos/WidgetsGallery/Demo/List/GroupedList/Vue/Light/) that has been reviewed and approved earlier. Addressing the data source as an API reference never seemed to be an issue. Moreover, it is an established pattern that can be seen throughout other demos. Therefore I have to respectfully disagree and insist on keeping the original version intact.

Reply:
Alright. You can keep the [dataSource] form, none argues about that. But "dataSource" is not a noun, it is a name, and you can't use an indefinite articles with proper names. Just omit the article then.

```suggestion
You can use [dataSource](/Documentation/ApiReference/UI_Components/dxTagBox/Configuration/#dataSource) where each entry is an object that contains the **key** and **items** fields. The **key** field specifies the group header (parent), and the **items** field holds an array of child items. A data source can specify different field names if the data is structured in the same way. In this case, implement the [DataSource](/Documentation/ApiReference/Data_Layer/DataSource/)'s [map](/Documentation/ApiReference/Data_Layer/DataSource/Configuration/#map) function to create **key** and **items** field mappings. Refer to the following help topic for more information: [Item Mapping](/Documentation/Guide/Data_Binding/Data_Layer/#Reading_Data/Data_Transformation/Item_Mapping).
```

Reply:
Let me chime in. We tend to treat property and component names as usual words ("data source") because that's how you subvocalize them when reading. And as a countable noun, the word "source" demands an article. However, we keep the original spelling of an API member (`dataSource`) because the docs are for developers, and that's how this member is named in code.

Reply:
Alright, if this is established practice for you, I have no further objections. Let's see what Dirk, Albert, or Armina has to say about it.

Reply:
I think whether to choose "dataSource" or "data source" is more of a technical decision, but I would like to make a suggestion. If you think it's better to use "dataSource", could we add a noun after this term that describes it? For example, if it's an API, would it be technically correct to say "a dataSource API"?
Let me know what you think.

Reply:
I'm afraid this also doesn't correspond with the overall style used throughout other demos. What I don't understand is why do we need to change anything? As I mentioned earlier, this paragraph is copied from a similar demo pretty much verbatim. That demo was reviewed by at least three people (including an editor), approved, and posted on the website. So what has changed since?

Reply:
Okay, if this format is used in our other demos, we can leave it as "dataSource" for consistency. Thank you.